src/helpers/events.js

```diff
@@ -435,11 +436,15 @@ exports.getContract = (contracts, startDate, endDate) => contracts.find((cont) =
 
 exports.workingStats = async (query, credentials) => {
   const companyId = get(credentials, 'company._id', null);
-  const queryAuxiliaries = { company: companyId };
-  if (query.auxiliary) {
-    const ids = UtilsHelper.formatObjectIdsArray(query.auxiliary);
-    queryAuxiliaries._id = { $in: ids };
-  }
+  const queryAuxiliaries = {};
+  let ids = [];
+
+  if (!query.auxiliary) {
+    const users = await UserCompany.find({ company: companyId }, { user: 1 });
+    for (const user of users) ids.push(user.user);
+  } else ids = UtilsHelper.formatObjectIdsArray(query.auxiliary);
+
+  queryAuxiliaries._id = { $in: ids };
```

Comment:
Que penses-tu de : 
```
let queryAuxiliaries;
if (query.auxiliary) { 
  queryAuxiliaries = { _id: { $in: UtilsHelper.formatObjectIdsArray(query.auxiliary) } };
} else {
  const userCompanies = await UserCompany.find({ company: companyId }, { user: 1 }).lean();
  queryAuxiliaries = { _id: { $in: users.map(user => user._id) } }
}
```

Reply:
Il faut ajouter un .lean à ton find sur UserCompany aussi

Reply:
Oui top, je rectifie!

Reply:
que pensez vous de changer comme suit pour mutualiser la partie avec le `{ _id: { $in` :

let auxiliaryIds = []
if (query.auxiliary) {
    auxiliaryIds = UtilsHelper.formatObjectIdsArray(query.auxiliary);
  } else {
    const users = await UserCompany.find({ company: companyId }, { user: 1 }).lean();
    auxiliaryIds = users.map(u => u.user);
  }

const auxiliaries = await User.find({ _id: { $in: auxiliaryIds } }).populate('contracts').lean();